chore: Cleanup native tests

pramodsatya opened the pull request:
## Description
Cleans up tests from `presto-native-tests`.

## Motivation and Context
Helper methods used to test custom functions are only used in `TestCustomFunctions` and should not be a part of `NativeTestUtils`.
Builder pattern should be used to get the `queryRunner` in tests and function `NativeTestUtils.createNativeQueryRunner` should not be overloaded with different arguments.
Enables tests with time type in `AbstractTestAggregationsNative`, since it is now supported in Velox.

## Impact
NA


```
== NO RELEASE NOTE ==
```

pramodsatya commented:
Thanks @aditi-pandit, I get the intent now and reverted the changes to `implicitCastToVarchar` flag. Could you PTAL?

pdabre12 reviewed (approved):
Thanks @pramodsatya. LGTM.

pramodsatya commented:
> Thanks @pramodsatya. LGTM.

Thanks @pdabre12, I extracted the `NativeTypeManager` change to a separate PR here: https://github.com/prestodb/presto/pull/26743, could you PTAL?

I had added the commit here just to verify the aggregation test, which was failing with `Unsupported type` error, runs successfully.

pramodsatya commented:
> Thanks @pramodsatya

Thanks @aditi-pandit, there were some test failures in CI and they are resolved by https://github.com/prestodb/presto/commit/5163ca4629636c9df87d797387d02359f61c59e8. Could you please review/approve again?
